File: Rubik.HTML/Server/Schema/Schema.cs
using System;
using System.Web;
using System.ComponentModel;
using System.Collections.Generic;

namespace Rubik.Server.Schema
{
    public class Schema
    {
        public Axis Columns
        {
            get; set;
        }

        public Axis Rows
        {
            get; set;
        }

        public Axis Filters
        {
            get; set;
        }

        public List<MeasureInfo> Data
        {
            get; set;
        }

        public List<CalculatedMember> CalculatedMembers
        {
            get; set;
        }

        public List<CalculatedSet> CalculatedSets
        {
            get; set;
        }

        public String CubeName
        {
            get; set;
        }
    }
}

File: Rubik.HTML/Lib/Schema/Schema.ts
/// <reference path="Interfaces.ts" />
/// <reference path="Axis.ts" />
/// <reference path="MeasureInfo.ts" />
/// <reference path="CalculatedMember.ts" />
/// <reference path="CalculatedSet.ts" />



module Rubik.Server.Schema {
    export class Schema implements ISchema {
        public Columns: Axis;
        public Rows: Axis;
        public Filters: Axis;
        public Data: List<MeasureInfo>;
        public CalculatedMembers: Rubik.Collections.List<CalculatedMember>;
        public CalculatedSets: Rubik.Collections.List<CalculatedSet>;
        public CubeName: string;
    }
}
